### DISCUSSION TITLE
Status code 204 must not have a response body when return use  NoReturn

### DISCUSSION THREAD

**Discussion by me-v2** [fastapi/fastapi#9336] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/event", summary="203", status_code=204)
async def audit_log_event(log: CreateEventSchema)->NoReturn:
  
    ::return code: 204
    """
    log = Event(**log.dict())
    session.add(log)
    await session.commit()
```

### Description

error ::

```
    assert is_body_allowed_for_status_code(
AssertionError: Status code 204 must not have a response body
```

### Operating System

macOS

### Operating System Details

noe 

### FastAPI Version

0.95.0

### Python Version

3.11.2

**Comment by Kludex** [comment, marked as answer]
`NoReturn` is not the correct type here... It should be `None`.

Ref.: https://docs.python.org/3/library/typing.html#typing.NoReturn
